client/varsdir.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2024 Broadcom, Inc. All Rights Reserved.
+ *
+ * Licensed under the GNU Lesser General Public License v2.1 (the "License");
+ * you may not use this file except in compliance with the License. The terms
+ * of the License are located in the COPYING file of this distribution.
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <dirent.h>
+#include <errno.h>
+
+#include "../llconf/nodes.h"
+
+struct cnfnode *parse_varsdirs(char *dirs[])
+{
+	struct cnfnode *cn_top = NULL;
+    DIR *fdir = NULL;
+    FILE *fptr = NULL;
+    int i;
+
+    cn_top = create_cnfnode("(root)");
+
+    for (i = 0; dirs[i]; i++) {
+        fdir = opendir(dirs[i]);
+        if (!fdir) {
+            if (errno == ENOENT)
+                /* ignore non-existing directories */
+                continue;
+            else
+                goto error;
+        }
+
+        struct dirent *dent;
+        while((dent = readdir(fdir))) {
+            char buf[256], path[256];
+            char *p;
+
+            p = dent->d_name;
+            /* skip disallowed filenames */
+            while(*p && ((*p == '_') || isdigit(*p) || islower(*p))) p++;
+            if (*p) {
+                continue;
+            }
+#pragma GCC diagnostic push
```

Comment:
I think so, since `PATH_MAX` has the same value, and this is also the size of `dent->d_name`. So the result can always be greater. It's okay to truncate, since the path name would be too long otherwise.